nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/SplitText.java

```diff
@@ -147,138 +193,200 @@ protected void init(final ProcessorInitializationContext context) {
+                return new EndOfLineMarker(bytesRead, eolBuffer, true, bytesToWriteFirst);  // bytesToWriteFirst should be "null"?
             }
 
-            // If we get a character that's not an end-of-line char, then
-            // we need to write out the EOL's that we have buffered (if out != null).
-            // Then, we need to reset our EOL buffer because we no longer have consecutive EOL's
-            if (!isNewLineChar) {
+            // Verify leading bytes do not violate size limitation
+            if (bytesToWriteFirst != null && (bytesToWriteFirst.length + bytesRead) > (maxSize - bytesReadSoFar) && includeLineDelimiter) {
+                return new EndOfLineMarker(-1, eolBuffer, false, leadingNewLineBytes);
+            }
+            // Write leadingNewLines, if appropriate
+            if ( buffer != null && includeLineDelimiter && bytesToWriteFirst != null) {
+                bytesRead += bytesToWriteFirst.length;
+                buffer.write(bytesToWriteFirst);
+                bytesToWriteFirst = null;
+            }
+            // buffer the output
+            bytesRead++;
+            if (buffer != null && nextByte != '\n' && nextByte != '\r') {
                 if (bytesToWriteFirst != null) {
-                    if (out != null) {
-                        out.write(bytesToWriteFirst);
-                    }
-
-                    bytesToWriteFirst = null;
-                }
-
-                if (out != null) {
-                    eolBuffer.drainTo(out);
+                    buffer.write(bytesToWriteFirst);
                 }
-
+                bytesToWriteFirst = null;
+                eolBuffer.drainTo(buffer);
                 eolBuffer.clear();
+                buffer.write(nextByte);
             }
 
-            // if there's an OutputStream to copy the data to, copy it, if appropriate.
-            // "if appropriate" means that it's not a line delimiter or that we want to copy line delimiters
-            bytesRead++;
-            if (out != null && (includeLineDelimiter || !isNewLineChar)) {
-                if (bytesToWriteFirst != null) {
-                    out.write(bytesToWriteFirst);
-                    bytesToWriteFirst = null;
+            // check the size limit
+            if (bytesRead > (maxSize-bytesReadSoFar) && bytesReadSoFar > 0) {
+                in.reset();
+                if (buffer != null) {
+                    buffer.close();
                 }
-
-                out.write(nextByte);
+                return new EndOfLineMarker(-1, eolBuffer, false, leadingNewLineBytes);
             }
 
             // if we have a new line, then we're done
             if (nextByte == '\n') {
-                eolBuffer.addEndOfLine(lastByte == '\r', true);
+                if (buffer != null) {
+                    buffer.writeTo(out);
+                    buffer.close();
+                    eolBuffer.addEndOfLine(false, true);  //TODO: verify "false" is equivalent to "lastByte == '\r'"
```

Comment:
Is this TODO a reminder to future @markobean?  Does a current unit test verify it?